rs/xnet/payload_builder/src/certified_slice_pool.rs

```diff
@@ -377,7 +377,18 @@ impl Payload {
         message_limit: Option<usize>,
         byte_limit: Option<usize>,
     ) -> CertifiedSliceResult<(Option<Self>, Option<Self>)> {
-        let message_limit = message_limit.unwrap_or(usize::MAX);
+        // `messages_end` may not exceed a certain `max_messages_index` in order to cap the number
```

Comment:
Could we change the comment above to something like:
> Takes a slice prefix whose estimated size meets the explicit limits below, as well as an implicit limit on the number of messages which ensures that the number of signals in the reverse stream stays bounded.

Reply:
Forgot to mention this in the comment, good catch.